File: StringCompression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int n = chars.size();
        int sum = 1;
        char c = chars[0];
        vector<char> v;
        v.push_back(c);
        string s;
        
        for(int i = 1 ; i<n ; i++)
        {
            if(c == chars[i])
                sum++;
            else
            {
                if(sum != 1)
                {
                    s = to_string(sum);
                    for(int j = 0; j< s.size() ; j++)
                        v.push_back(s[j]);
                    v.push_back(chars[i]);
                    sum = 1;
                }
                else
                    v.push_back(chars[i]);
                c = chars[i];
                
            }
            
        }
        
        s = to_string(sum);
        if(sum != 1)
        {
            for(int j = 0; j< s.size() ; j++)
                        v.push_back(s[j]);
        }
        
        
        chars = v;
        return v.size();
    }
};

File: StringCompression.py
class Solution:
    def compress(self, chars: List[str]) -> int:

        first = 0
        second = 0
        size = 0

        while second < len(chars):
            if chars[first] != chars[second]:
                if second - first > 1:
                    length = str(second - first)
                    for c in length:
                        first += 1
                        chars[first] = c
                        size += 1
        
                size += 1

                first += 1
                while first != second:
                    chars[first] = ""
                    first += 1

            second += 1

        if second - first > 1:
            length = str(second - first)
            for c in length:
                first += 1
                chars[first] = c
                size += 1

        size += 1
        
        left = 0
        right = 0

        while right < len(chars):
            if chars[right] != "":
                chars[left], chars[right] = chars[right], chars[left]
                left += 1
    
            right += 1

        return size
